wolfssl/wolfcrypt/sp_int.h

```diff
@@ -747,6 +740,86 @@ typedef struct sp_ecc_ctx {
 /* Mask of all used bits in word/digit. */
 #define MP_MASK    SP_MASK
 
+#ifdef MP_LOW_MEM
+/* Use algorithms that use less memory. */
+#define WOLFSSL_SP_LOW_MEM
+#endif
+
+
+/* The number of bytes to a sp_int with 'cnt' digits.
+ * Must have at least one digit.
+ */
+#define MP_INT_SIZEOF(cnt)                                              \
+    (sizeof(sp_int) - (SP_INT_DIGITS - (((cnt) == 0) ? 1 : (cnt))) *    \
+     sizeof(sp_int_digit))
+/* The address of the next sp_int after one with 'cnt' digits. */
+#define MP_INT_NEXT(t, cnt) \
+    (sp_int*)(((byte*)(t)) + MP_INT_SIZEOF(cnt))
+
+
+/* Calculate the number of words required to support a number of bits. */
+#define MP_BITS_CNT(bits)                                       \
+        (((bits + SP_WORD_SIZE - 1) / SP_WORD_SIZE) * 2 + 1)
+
+#ifdef WOLFSSL_SMALL_STACK
+/*
+ * Dynamic memory allocation of mp_int.
+ */
+/* Declare a dynamically allocated mp_int. */
+#define DECL_MP_INT_SIZE_DYN(name, bits, max)                               \
+    sp_int* name = NULL
+/* Declare a dynamically allocated mp_int. */
+#define DECL_MP_INT_SIZE(name, bits)                                        \
+    sp_int* name = NULL
+/* Allocate an mp_int of minimal size and zero out. */
+#define NEW_MP_INT_SIZE(name, bits, heap, type)                             \
+do {                                                                        \
+    name = (mp_int*)XMALLOC(MP_INT_SIZEOF(MP_BITS_CNT(bits)), heap, type);  \
+    if (name != NULL) {                                                     \
+        XMEMSET(name, 0, MP_INT_SIZEOF(MP_BITS_CNT(bits)));                 \
+    }                                                                       \
+}                                                                           \
+while (0)
+/* Dispose of dynamically allocated mp_int. */
+#define FREE_MP_INT_SIZE(name, heap, type) \
+    XFREE(name, heap, type)
+/* Type to cast to when using size marcos. */
+#define MP_INT_SIZE     sp_int
+/* Must check mp_int pointer for NULL. */
+#define MP_INT_SIZE_CHECK_NULL
```

Comment:
Does checking for this MP_INT_SIZE_CHECK_NULL macro need to happen with all mp_int creations now (i.e. asn.c / internal.c too).

Reply:
Only when using the macro around MP_INT_SIZE.
So existing code doesn't change.